client/X11/xf_rail.c

```diff
@@ -327,14 +327,14 @@ static BOOL xf_rail_window_common(rdpContext* context,
 			return FALSE;
 		}
 
-		HashTable_Add(xfc->railWindows, (void*)(UINT_PTR) orderInfo->windowId,
+		HashTable_Add(xfc->railWindows, (void*)(UINT_PTR) orderInfo->windowId + 1,
 		              (void*) appWindow);
```

Comment:
My suggestion would be to do
```C
HashTable_Add(xfc->railWindows, &appWindow->windowId, (void*) appWindow);
```
plus you add a comparison function on the `railWindows` that compares `UINT32 *`.
and getting values becomes
```C
return (xfAppWindow*) HashTable_GetItemValue(xfc->railWindows, &windowId);
```

Reply:
@hardening ok, you got me :+1 missed the obvious solution because there are local variables used when finding existing windows.